### PR TITLE
Add Sentry for production logging debugging

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

We have no visibility and we get no notifications when stuff goes wrong on our deployed applications.

Sentry can provide real-time Slack notifications when errors occur, and it will log the context of a request as well as the full error message.

### Changes proposed in this pull request

This PR adds:

- Raven the Ruby client for Sentry to the Gemfile
- configuration in the Rails production environment
- a `SENTRY_DSN` environment variable

### Guidance to review

Has the environment variable been unnecessarily added anywhere?

Also, apologies my editor has done some deletion of whitespaces so would recommend viewing diff to hide whitespace changes (Theo taught me dis).

### Link to Trello card

[82 - Add Sentry for production logging debugging](82-add-sentry-for-production-logging-debugging)

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -113,3 +113,7 @@
   config.hosts << ENV.fetch('DOMAIN')
   config.hosts << ENV.fetch('STAGING_DOMAIN') # Domain used in Azure pipelines for deploying to the second slot
 end
+
+Raven.configure do |config|
+  config.dsn = ENV.fetch('SENTRY_DSN')
```
Comment (tvararu) [inline config/environments/production.rb]: Non-blocking: looking at `manage-courses-frontend`, I don't think configuration is needed; Sentry seems to pick it up on its own when that's the env variable used.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -223,3 +228,5 @@ stages:
 #       railsEnv: 'production'
```
Comment (alexdesi) [inline azure-pipelines.yml]: could we remove all this bunch of comments 🤔 ?

File: config/environments/production.rb (hunk 1 above)
Comment (chubberlisk) [reply config/environments/production.rb]: Ah cool, since it's redundant I'll remove it!

File: azure-pipelines.yml (hunk 2 above)
Comment (chubberlisk) [reply azure-pipelines.yml]: I think this is the code for deploying our production app which we'll have at some point. Best to ask @tpinney.

**Pull request merged**